pkg/controller/cluster/controller.go

```diff
@@ -97,6 +97,9 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
 			klog.Infof("failed to add finalizer to cluster object %v due to error %v.", name, err)
 			return reconcile.Result{}, err
 		}
+
+		// Since adding the finalizer updates the object return to avoid later update issues
+		return reconcile.Result{}, nil
```

Comment:
I think you must set `Requeue: true` on the `reconcile.Result` in order to have the object reqd

Reply:
Is that actually the case here? Shouldn't the informer trigger another reconcile from the update?

Reply:
ah ofc yes, you are right